packages/gatsby/src/utils/start-server.ts

```diff
@@ -118,6 +118,67 @@ export async function startServer(
 
   const compiler = webpack(devConfig)
 
+  if (process.env.GATSBY_EXPERIMENT_LAZY_DEVJS) {
+    const bodyParser = require(`body-parser`)
+    const { boundActionCreators } = require(`../redux/actions`)
+    const { createClientVisitedPage } = boundActionCreators
+    app.use(bodyParser.json())
+    // Listen for the client marking a page as visited (meaning we need to
+    // compile its page component.
+    const chunkCalls = new Set()
+    app.post(`/___client-page-visited`, (req, res, next) => {
```

Comment:
can we add `bodyParser.json()` middleware here instead of adding global one? ( I don't think adding global one will break things, it just seems wasteful for all the other requests?)
```suggestion
    app.post(`/___client-page-visited`, bodyParser.json(), (req, res, next) => {
```

We use this per request handler middleware already in https://github.com/gatsbyjs/gatsby/blob/6858f22fc9a64f71756d5cd5122fab99d12749cf/packages/gatsby/src/utils/telemetry-server.ts#L44

Reply:
👍  makes sense